pandas/_libs/window.pyx

```diff
@@ -1241,49 +1242,69 @@ cdef _roll_min_max(ndarray[numeric] input, int64_t win, int64_t minp,
         input, win,
         minp, index, closed)
 
-    output = np.empty(N, dtype=input.dtype)
+    output = np.empty(N, dtype=float)
```

Comment:
Is this coercing integer input to float?

Reply:
yes, for two reasons: 
1. because when `closed=left` the first output should be NaN. 
2. because output elements up to minp should also be NaN.

I believe this behavior is consistent with rolling sum (coerced to float).